### PR TITLE
ExpressionFilterOperator NULL support.

### PR DESCRIPTION
**Pull request by shenyu0127**
This PR enables ExpressionFilterOperator NULL support.

We do not override `getNulls` because `getNulls` is called only by `BaseFilterOperator::getFalses` and we override the `getFalses`.

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/ExpressionScanDocIdIterator.java [hunk 1]
```diff
@@ -85,7 +88,7 @@ public int next() {
       ProjectionBlock projectionBlock =
           new ProjectionOperator(_dataSourceMap, new RangeDocIdSetOperator(blockStartDocId, _blockEndDocId))
               .nextBlock();
-      RoaringBitmap matchingDocIds = new RoaringBitmap();
+      MutableRoaringBitmap matchingDocIds = new MutableRoaringBitmap();
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/ExpressionScanDocIdIterator.java:91]: `RoaringBitmap` performs better than `MutableRoaringBitmap`. We should change all of them to use `RoaringBitmap`

File: pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/ExpressionScanDocIdIterator.java [hunk 2]
```diff
@@ -212,8 +215,27 @@ private void processProjectionBlock(ProjectionBlock projectionBlock, BitmapDataP
           default:
             throw new IllegalStateException();
         }
+        if (_nullHandlingEnabled) {
+          RoaringBitmap nullBitmap = _transformFunction.getNullBitmap(projectionBlock);
+          if (_predicateEvaluationResult == PredicateEvaluationResult.TRUE) {
+            if (nullBitmap != null) {
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/ExpressionScanDocIdIterator.java:221]: Also check if it is empty?

File: pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/ExpressionScanDocIdIterator.java [hunk 3]
```diff
@@ -212,8 +215,27 @@ private void processProjectionBlock(ProjectionBlock projectionBlock, BitmapDataP
...
+              RoaringBitmap fullBitmap = new RoaringBitmap();
+              fullBitmap.add(0L, (long) numDocs);
+              nullBitmap.xor(fullBitmap);
+              matchingDocIds.and(nullBitmap.toMutableRoaringBitmap());
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/ExpressionScanDocIdIterator.java:225]: It is the same as `andNot()`

File: pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/ExpressionScanDocIdIterator.java [hunk 4]
```diff
@@ -212,8 +215,27 @@ private void processProjectionBlock(ProjectionBlock projectionBlock, BitmapDataP
           default:
             throw new IllegalStateException();
         }
+        if (_nullHandlingEnabled) {
+          RoaringBitmap nullBitmap = _transformFunction.getNullBitmap(projectionBlock);
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/ExpressionScanDocIdIterator.java:219]: (MAJOR) What is the offset of this bitmap? Is this the docId within the projection block?

Comment (shenyu0127) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/ExpressionScanDocIdIterator.java]: Good point. Fixed the bug.

File: pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/ExpressionScanDocIdIterator.java (hunk 2 above)
Comment (shenyu0127) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/ExpressionScanDocIdIterator.java]: This comment no longer applies. I now have an empty check in the `isNull` function.

File: pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/ExpressionScanDocIdIterator.java (hunk 3 above)
Comment (shenyu0127) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/ExpressionScanDocIdIterator.java]: This comment no longer applies.

File: pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/ExpressionScanDocIdIterator.java (hunk 1 above)
Comment (shenyu0127) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/ExpressionScanDocIdIterator.java]: Reverted the change.

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**